components/merkledb/src/migration.rs

```diff
@@ -429,40 +429,49 @@ impl fmt::Debug for MigrationHelper {
 impl MigrationHelper {
     /// Creates a new helper.
     pub fn new(db: impl Into<Arc<dyn Database>>, namespace: &str) -> Self {
-        let (this, handle) = Self::with_handle(db, namespace);
-        handle.forget();
-        this
+        assert_valid_name_component(namespace);
```

Comment:
Since this structure is public, wouldn't it be better to return `Result<Self, failure::Error>` here?

Reply:
The current implementation assumes that `namespace` comes from the trusted source and was checked previously. For example, in the Exonum context it is a service name, which is validated in advance (when the service is created).